gobblin-metastore/src/main/java/gobblin/metastore/DatabaseJobHistoryStore.java

```diff
@@ -318,194 +265,206 @@ public void close()
+          }
+          upsertMetric(upsertStatement, metric, taskExecution.getTaskId());
+          if (batchSize++ > 1000) {
+            executeBatches(upsertStatement);
+            upsertStatement = null;
+          }
+        }
+      }
+    }
+    executeBatches(upsertStatement);
   }
 
-  private void updateMetric(Connection connection, String template, String id, Metric metric, boolean insert)
-      throws SQLException {
+  private void upsertMetric(PreparedStatement upsertStatement, Metric metric, String id) throws SQLException {
     Preconditions.checkArgument(!Strings.isNullOrEmpty(id));
     Preconditions.checkArgument(metric.hasGroup());
     Preconditions.checkArgument(metric.hasName());
     Preconditions.checkArgument(metric.hasType());
     Preconditions.checkArgument(metric.hasValue());
 
-    PreparedStatement updateStatement = connection.prepareStatement(template);
     int index = 0;
-    if (insert) {
-      updateStatement.setString(++index, id);
-      updateStatement.setString(++index, metric.getGroup());
-      updateStatement.setString(++index, metric.getName());
-      updateStatement.setString(++index, metric.getType().name());
-      updateStatement.setString(++index, metric.getValue());
-    } else {
-      updateStatement.setString(++index, metric.getValue());
-      updateStatement.setString(++index, id);
-      updateStatement.setString(++index, metric.getGroup());
-      updateStatement.setString(++index, metric.getName());
-      updateStatement.setString(++index, metric.getType().name());
-    }
-    updateStatement.executeUpdate();
+    upsertStatement.setString(++index, id);
+    upsertStatement.setString(++index, metric.getGroup());
+    upsertStatement.setString(++index, metric.getName());
+    upsertStatement.setString(++index, metric.getType().name());
+    upsertStatement.setString(++index, metric.getValue());
+    upsertStatement.addBatch();
   }
 
-  private boolean existsProperty(Connection connection, String template, String id, String key)
-      throws SQLException {
-    Preconditions.checkArgument(!Strings.isNullOrEmpty(id));
-    Preconditions.checkArgument(!Strings.isNullOrEmpty(key));
-
-    PreparedStatement queryStatement = connection.prepareStatement(template);
-    int index = 0;
-    queryStatement.setString(++index, id);
-    queryStatement.setString(++index, key);
-    return queryStatement.executeQuery().next();
+  private void executeBatches(PreparedStatement upsertStatement) throws SQLException {
+    if (upsertStatement != null) {
+      try {
+        upsertStatement.executeBatch();
+      } finally {
+        upsertStatement.close();
+      }
+    }
   }
 
-  private void updateProperty(Connection connection, String template, String id, String key, String value,
-      boolean insert)
+  private void upsertTaskExecutionInfo(PreparedStatement upsertStatement, TaskExecutionInfo info)
```

Comment:
Can you rename this method and the corresponding methods for job props, job metrics, etc. to something along the lines of `addTaskExecInfoToBatch` - the current method sounds like the upsert is actually getting executed.